src/components/general/EntryVerify/index.tsx

```diff
@@ -117,16 +121,11 @@ function EntryVerify(props: Props) {
         setEntryVerificationRequest,
     } = requests;
 
-    // FIXME: we can remove verified and use value instead
-    const [verified, setVerificationStatus] = useState(value);
-
-    useEffect(() => {
-        setVerificationStatus(value);
-    }, [value]);
-
-    // FIXME: memoize this
-    const selectedOption = verificationStatusOptions.find(v => v.key === verified) ||
-                           verificationStatusOptions[1];
+    const selectedOption = useMemo(
+        () => verificationStatusOptions.find(v => v.key === value) ??
+            verificationStatusOptions[1],
```

Comment:
We should think about a better way to identify a default value @thenav56 @frozenhelium @AdityaKhatri

Reply:
At least, it shouldn't break silently.